### ISSUE TITLE
Warn and check out if a downloaded binary can't be used

### ISSUE THREAD

**Issue by diederikh** [Carthage/Carthage#981]
I have the following Cartfile:

```
github "firebase/FirebaseUI-iOS"
```

When doing a `carthage update` the log says:

```
*** Fetching FirebaseUI-iOS
*** Downloading FirebaseUI-iOS.framework binary at "FirebaseUI 0.3.1 Release"
*** xcodebuild output can be found in /var/folders/56/hdkhtvl52y3dh7lfdhq4yg9r0000gp/T/carthage-xcodebuild.BMyjSM.log
```

However the framework is not downloaded to the build folder. 
When I look at the 0.3.1 release in Github is does have a `FirebaseUI.framework.zip` file attached to the release.

**Comment by mdiep** [comment]
Carthage expects the framework to be inside a `Carthage/Build/$PLATFORM/` directory inside the `.zip`. It looks like that's not the case here. (You can create this by executing `carthage build --no-skip-current && carthage archive <PROJECT-NAME>` inside the repository.)

**Comment by mdiep** [comment]
In the meantime, you can build FirebaseUI manually by using the `--no-use-binaries` flag.

**Comment by mdiep** [comment]
Re-opening this because Carthage should do something to help out if it downloads a file that it can't use.

**Comment by ikesyo** [comment]
Maybe this can be closed now by #1755.
